pkg/authorization/apiserver/admission/restrictusers/restrictusers.go

```diff
@@ -167,19 +163,17 @@ func (q *restrictUsersAdmission) Validate(a admission.Attributes) (err error) {
 		return nil
 	}
 
-	// TODO: Cache rolebinding restrictions.
-	roleBindingRestrictionList, err := q.roleBindingRestrictionsGetter.RoleBindingRestrictions(ns).
-		List(metav1.ListOptions{})
+	roleBindingRestrictionList, err := q.rbrLister.List(labels.Everything())
 	if err != nil {
 		return admission.NewForbidden(a, err)
 	}
-	if len(roleBindingRestrictionList.Items) == 0 {
+	if len(roleBindingRestrictionList) == 0 {
```

Comment:
need a `HasSynced` somewhere (see where other admission plugins do that)

Reply:
added